src/wasm2asm.h

```diff
@@ -655,7 +655,8 @@ Ref Wasm2AsmBuilder::processFunction(Function* func) {
   ExpressionList* stats = isBodyBlock ?
       &static_cast<Block*>(func->body)->list : nullptr;
   bool endsInReturn =
-      (isBodyBlock && ((*stats)[stats->size()-1]->_id == Expression::ReturnId));
+      (isBodyBlock && ((*stats)[stats->size()-1]->_id == Expression::ReturnId)) ||
```

Comment:
While you're here, please change `->_id == Expression::ReturnId` to `->is<Return>()`. (Also the same happens with `Block` a few lines above.)